pkgs/by-name/no/nordvpn/package.nix

```diff
@@ -0,0 +1,148 @@
+  iptables,
+  lib,
+  libidn2,
+  libxml2,
+  pkg-config,
+  procps,
+  sysctl,
+  wireguard-tools,
+  zlib,
+}:
+
+let
+  baseURL = "https://github.com/Tunnelblick/Tunnelblick/raw/master/third_party/sources/openvpn";
+  patchedPkgs = appendOverlays [
+    (final: prev: {
+      # Nordvpn uses a patched openvpn in order to perform xor obfuscation
+      # See https://github.com/NordSecurity/nordvpn-linux/blob/e614303aaaf1a64fde5bb1b4de1a7863b22428c4/ci/openvpn/check_dependencies.sh
+      openvpn = prev.openvpn.overrideAttrs (old: {
+        patches =
+          (old.patches or [ ])
+          ++ (
+            let
+              patchedVersion = "2.6.14";
+              patchURL = "${baseURL}/openvpn-${patchedVersion}/patches";
+            in
+            [
+              (prev.fetchpatch {
+                url = "${patchURL}/02-tunnelblick-openvpn_xorpatch-a.diff";
+                hash = "sha256-b9NiWETc0g2a7FNwrLaNrWx7gfCql7VTbewFu3QluFk=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/03-tunnelblick-openvpn_xorpatch-b.diff";
+                hash = "sha256-X/SshB/8ItLFBx6TPhjBwyA97ra0iM2KgsGqGIy2s9I=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/04-tunnelblick-openvpn_xorpatch-c.diff";
+                hash = "sha256-fw0CxJGIFEydIVRVouTlD1n275eQcbejUdhrU1JAx7g=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/05-tunnelblick-openvpn_xorpatch-d.diff";
+                hash = "sha256-NLRtoRVz+4hQcElyz4elCAv9l1vp4Yb3/VJef+L/FZo=";
+              })
+              (prev.fetchpatch {
+                url = "${patchURL}/06-tunnelblick-openvpn_xorpatch-e.diff";
+                hash = "sha256-mybdjCIT9b6ukbGWYvbr74fKtcncCtTvS5xSVf92T6Y=";
+              })
+            ]
+          );
+      });
+    })
+  ];
+in
+buildGoModule rec {
+  pname = "nordvpn";
+  version = "3.20.2";
+
+  src = fetchFromGitHub {
+    owner = "NordSecurity";
+    repo = "nordvpn-linux";
+    rev = version;
+    sha256 = "sha256-XB/wOdp97VnAkzHTHrJ2xffEx4HNFWn9yjFz3TsaEsE=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    cacert
+    gcc
+    iproute2
+    iptables
+    libidn2
+    libxml2
+    procps
+    sysctl
+    zlib
+  ];
+
+  vendorHash = "sha256-gW/KieixLS6TTcBOHeYK9L8sieERrj6ia+iov8rT2AE=";
+
+  ldflags = [
+    "-X main.Version=${version}"
+    "-X main.Environment=dev"
+    "-X main.Salt=development"
+    "-X main.Hash=${src.rev}"
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    echo "Building nordvpn CLI..."
+    sed -i "s|/usr/lib/nordvpn|$out/bin|g" internal/constants.go
+
+    old_ovpn_path='filepath.Join(internal.AppDataPathStatic, "openvpn")'
+    new_ovpn_path=\"${patchedPkgs.openvpn}/bin/openvpn\"
+    sed -i "s|$old_ovpn_path|$new_ovpn_path|" daemon/vpn/openvpn/config.go
+
+    export LDFLAGS="${builtins.concatStringsSep " " ldflags}"
+    go build -ldflags "$LDFLAGS" -o bin/nordvpn ./cmd/cli
```

Comment:
The `buildGoModule` wrapper is made to build go app. I see that here you're overriding the `buildPhase`, hindering the wrapper mechanism. Wouldn't it be possible to avoid using the `buildPhase` and use exclusively `buildGoModule`?